src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -506,10 +506,26 @@ uint IdealLoopTree::estimate_peeling(PhaseIdealLoop *phase) {
   if (_head->is_CountedLoop()) {
     CountedLoopNode* cl = _head->as_CountedLoop();
     if (cl->is_unroll_only() || cl->trip_count() == 1) {
+      // Peeling is not legal here (cf. assert in do_peeling), we don't even stress peel!
       return 0;
     }
   }
 
+#ifndef PRODUCT
+  // It is now safe to peel or not.
+  if (StressLoopPeeling) {
+    LoopNode* loop_head = _head->as_Loop();
+    static constexpr uint max_peeling_opportunities = 5;
+    if (loop_head->_peeling_opportunities_count < max_peeling_opportunities) {
+      loop_head->_peeling_opportunities_count++;
+      // In case of stress, let's just pick randomly...
+      return phase->C->random() % 2 == 0 ? estimate : 0;
+    }
```

Comment:
Nice, looks much better :-)

`_peeling_opportunities_count` could be confused with real opportunities without stressing. Maybe we can rename it to `_stress_peeling_opportunities`?